lineapy/api/api.py

```diff
@@ -200,6 +200,19 @@ def get(artifact_name: str, version: Optional[int] = None) -> LineaArtifact:
     return linea_artifact
 
 
+def reload() -> None:
```

Comment:
I like this active reloading approach.

@mingjerli has already implemented some config handling in the middle of the execution in the ipython handler https://github.com/LineaLabs/lineapy/blob/main/lineapy/editors/ipython.py#L119-L130, and I think we should refactor it to be logic similar to this reload. I think it will be easier to understand and maintain.